embassy-nrf/src/pwm.rs

```diff
@@ -55,14 +55,14 @@ impl<'d, T: Instance> SequencePwm<'d, T> {
     /// mechanisms) on stack allocated buffers which which have been passed to
     /// [`new()`](SequencePwm::new).
     #[allow(unused_unsafe)]
-    pub fn new<'a>(
+    pub fn new(
         _pwm: impl Unborrow<Target = T> + 'd,
         ch0: impl Unborrow<Target = impl GpioOptionalPin> + 'd,
         ch1: impl Unborrow<Target = impl GpioOptionalPin> + 'd,
         ch2: impl Unborrow<Target = impl GpioOptionalPin> + 'd,
         ch3: impl Unborrow<Target = impl GpioOptionalPin> + 'd,
         config: SequenceConfig,
-        sequence: &'a [u16],
+        sequence: &mut [u16],
```

Comment:
it doesn't have to be `mut`, sorry. I screwed up in my last comment. :see_no_evil: 

Also, it has to live for `'d` so that it's guaranteed to be live while DMA runs.